lib/scrub_fields.rb

```diff
@@ -94,24 +79,32 @@ def ocn(str)
     str.strip!
     candidates = str.split(OCN_SPLIT_DELIM)
 
+    # TODO: maybe we don't care about this?
+    # I've found it to be a good indicator that the record wasn't
+    # exported properly, but there are legit cases of this I guess.
     if candidates.size > MAX_NUM_ITEMS
       count_x(:too_many_ocns)
-      @logger.error "Too many items (#{candidates.size}) in ocn #{str}"
+      Services.scrub_logger.warn "Too many ocns (#{candidates.size}) in ocn #{str}"
+    end
+
+    if candidates.size > 1
+      count_x(:multi_ocn_record)
     end
 
-    # DO WHILE MAYBE COMEFROM reject_value
+    # PLEASE DO WHILE MAYBE COMEFROM reject_value
     candidates.each do |candidate|
       catch(:rejected_value) do
         numeric_part = capture_numeric(candidate)
         candidate.strip!
 
-        # Try to find a reason to reject the candidate ocn
-        # Any time reject_value triggers, go ^^ to the catch.
+        # Try to find a reason to reject the candidate ocn.
+        # Any time reject_value triggers, it throws :rejected_value
+        # and we go back up to the catch statement inside the each-loop.
         candidate.nil? &&
-          reject_value("ocn is nil", "")
+          reject_value("ocn is nil")
```

Comment:
I haven't seen `throw`/`catch` in Ruby before and had to look it up. Curious if others have seen it before. I don't think we're using the return value stuff that `catch` does, so could we just do something like `log_why_it_is_bad && next`?

Reply:
I was under the impression that `next` only works inside the current block. So especially now that I've broken out a big chunk of the ocn method, a lot of the potential rejections happen in another block.

In perl I would have labeled the loop I want to return to as "foo" and said `next foo`, and this is the closest I've come to it in Ruby. I like that I don't have to sprinkle `next` everywhere in the code too.

Reply:
Essentially I'm suggesting (inlining the relevant part of `reject_value`)

```ruby
if candidate.nil? 
  count_x("value rejected: ocn is nil")
  next
end
```

We could make that a little terser by doing e.g.

```ruby
next if candidate.nil? && count_x("value rejected: ocn is nil")
```

or you could extract methods and do something like 
```ruby
next if candidate_nil?
```
and extract a method along the lines of:
```ruby
def candidate_nil?
  if candidate.nil?
    count_x("value rejected: ocn is nil")
  end
```
(this will return the value of `count_x(...)`, but you could always explicitly return true there if necessary

Would that work, or am I missing something? I'm not sure the conciseness is worth the (to me) obscure construct -- let's discuss tomorrow